mlrun/datastore/google_cloud_storage.py

```diff
@@ -84,7 +51,30 @@ def get_filesystem(self, silent=True):
         return self._filesystem
 
     def get_storage_options(self):
-        return dict(token=self._get_secret_or_env("GOOGLE_APPLICATION_CREDENTIALS"))
+        google_credentials = self._get_secret_or_env("GOOGLE_APPLICATION_CREDENTIALS")
+        gcp_credentials = self._get_secret_or_env("GCP_CREDENTIALS")
+        if (
+            gcp_credentials
+            and google_credentials
+            and google_credentials != gcp_credentials
+        ):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Both GOOGLE_APPLICATION_CREDENTIALS and GCP_CREDENTIALS are set but not the same"
+            )
+        credentials = google_credentials or gcp_credentials
+        if credentials:
+            try:
+                credentials = json.loads(credentials)
```

Comment:
why do you need to use try and except?

Reply:
json.loads will not work on a path to json file, as GOOGLE_APPLICATION_CREDENTIALS.